services/blockchain-indexer/shared/indexer/transactionProcessor/pos/stake.js

```diff
@@ -90,29 +93,71 @@ const decrementStakeTrx = async (stake, trx) => {
 	await stakesTable.decrement(decrementParam, trx);
 };
 
+const updateTotalStake = async (changeAmount, dbTrx) => {
+	const tokenID = await getPosTokenID();
+	const tokenKey = KV_STORE_KEYS.PREFIX.TOTAL_STAKED.concat(tokenID);
+	const curStakedAmount = BigInt(await keyValueTable.get(tokenKey) || 0);
+	const newStakedAmount = curStakedAmount + changeAmount;
+
+	await keyValueTable.set(tokenKey, newStakedAmount, dbTrx);
+};
+
+const updateTotalSelfStake = async (changeAmount, dbTrx) => {
+	const tokenID = await getPosTokenID();
+	const tokenKey = KV_STORE_KEYS.PREFIX.TOTAL_SELF_STAKED.concat(tokenID);
+	const curStakedAmount = BigInt(await keyValueTable.get(tokenKey) || 0);
+	const newStakedAmount = curStakedAmount + changeAmount;
+
+	await keyValueTable.set(tokenKey, newStakedAmount, dbTrx);
+};
+
 // eslint-disable-next-line no-unused-vars
 const applyTransaction = async (blockHeader, tx, dbTrx) => {
 	const stakes = await getStakeIndexingInfo(tx);
+	let totalStakeChange = BigInt(0);
+	let totalSelfStakeChange = BigInt(0);
 
-	logger.trace(`Indexing transaction ${tx.id} contained in block at height ${tx.height}.`);
+	logger.trace(`Indexing stakes in transaction ${tx.id} contained in block at height ${tx.height}.`);
 	await BluebirdPromise.map(
 		stakes,
-		async (stake) => incrementStakeTrx(stake, dbTrx),
+		async (stake) => {
+			await incrementStakeTrx(stake, dbTrx);
+
+			if (stake.stakerAddress === stake.validatorAddress) {
```

Comment:
Please double check this. I don't think `stakes` have the `stakerAddress` property. You'll most likely have to extract the senderAddress from the senderPublicKey, if this information is not directly available.

Please reach out if you need to extract the `senderAddress` to discuss further specifics.

Reply:
this works because we format the transaction to generate stakerAddress from publicKey.

https://github.com/LiskHQ/lisk-service/blob/527fd1de9646b9e04b1973438cfaf9eb97f9fe86/services/blockchain-indexer/shared/indexer/transactionProcessor/pos/stake.js#L116